go/libraries/doltcore/diff/schema_diff.go

```diff
@@ -60,31 +59,33 @@ func DiffSchemas(sch1, sch2 schema.Schema) (map[uint64]SchemaDifference, error)
 		}
 	}
 
-	return diffs, nil
+	return diffs, unionTags
 }
 
 // pairColumns loops over both sets of columns pairing columns with the same tag.
-func pairColumns(sch1, sch2 schema.Schema) (map[uint64][2]*schema.Column, error) {
-	colPairMap := make(map[uint64][2]*schema.Column)
-	err := sch1.GetAllCols().Iter(func(tag uint64, col schema.Column) (stop bool, err error) {
-		colPairMap[tag] = [2]*schema.Column{&col, nil}
+func pairColumns(sch1, sch2 schema.Schema) (map[uint64]columnPair, []uint64) {
+	// collect the tag union of the two schemas, ordering sch1 before sch2
+	var unionTags []uint64
+	colPairMap := make(map[uint64]columnPair)
+
+	_ = sch1.GetAllCols().Iter(func(tag uint64, col schema.Column) (stop bool, err error) {
+		colPairMap[tag] = columnPair{&col, nil}
+		unionTags = append(unionTags, tag)
+
 		return false, nil
 	})
 
-	if err != nil {
-		return nil, err
-	}
-
-	err = sch2.GetAllCols().Iter(func(tag uint64, col schema.Column) (stop bool, err error) {
+	_ = sch2.GetAllCols().Iter(func(tag uint64, col schema.Column) (stop bool, err error) {
 		if pair, ok := colPairMap[tag]; ok {
 			pair[1] = &col
 			colPairMap[tag] = pair
 		} else {
-			colPairMap[tag] = [2]*schema.Column{nil, &col}
+			colPairMap[tag] = columnPair{nil, &col}
```

Comment:
@bheni I added this change since the last review. 
Previously we were really on columns to have tags in ascending order when displaying SchemaDiffs. I refactored this method to respect the column order of the schemas and to return the ordered union of tags.